hack/dockerfiles/generated-files.Dockerfile

```diff
@@ -0,0 +1,29 @@
+# protoc is dynamically linked to glibc to can't use golang:1.9-alpine
+FROM golang:1.9 AS gobuild-base
+ARG PROTOC_VERSION=3.1.0
+ARG GOGO_VERSION=master
+RUN apt-get update && apt-get install -y \
+ git \
+ unzip \
+ && true
+RUN wget -q https://github.com/google/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip && unzip protoc-${PROTOC_VERSION}-linux-x86_64.zip -d /usr/local
+RUN protoc --version
+
+RUN go get -d github.com/gogo/protobuf/protoc-gen-gogofaster \
+  && cd /go/src/github.com/gogo/protobuf \
+        && git checkout -q $GOGO_VERSION \
+        && go install ./protoc-gen-gogo ./protoc-gen-gogofaster ./protoc-gen-gogoslick
+
+WORKDIR /go/src/github.com/moby/buildkit
+COPY . .
+RUN go generate ./...
+
+RUN mkdir /generated-files
+RUN find . -name "*.pb.go" ! -path ./vendor/\* | tar -cf - --files-from - | tar -C /generated-files -xf -
+RUN git status --porcelain -- **/*.pb.go 2>/dev/null | tee /dev/stderr >/generated-files.status
+
+FROM scratch AS update
+
+COPY --from=gobuild-base generated-files /generated-files
```

Comment:
why the `generated-files` subdir, and therefore the `--strip-components` on extraction?

Reply:
From the commit message quoted in https://github.com/moby/buildkit/pull/323#issuecomment-376470317:
> Use a subdirectory and `tar --strip-components`
>    (portable to MacOS and Linux according to `tar(1)`) since trying to do a
>    `docker export` of just the root ends up adding `.dockerenv`, `sys`, `proc` and
>    `dev` to the source tree.

Reply:
Ok, didn't think of that. Lets leave it like it is then until we can switch to the local exporter that doesn't have this issue.

Reply:
You can add a comment about it in this Dockerfile as well.

Reply:
I added a comment before I went home last night and I've rewritten it this morning to make more sense...